pkg/server/status.go

```diff
@@ -849,6 +862,44 @@ func (s *statusServer) Nodes(
 			return nil, grpcstatus.Errorf(codes.Internal, err.Error())
 		}
 	}
+
+	// get the derivatives here?
+	var queries []tspb.Query
+	for _, nodeStatus := range resp.Nodes {
+		for _, metric := range IoMetrics {
+			deriv := tspb.TimeSeriesQueryDerivative_NON_NEGATIVE_DERIVATIVE
+			query := tspb.Query{
+				Name:       "cr.node." + metric,
+				Sources:    []string{nodeStatus.Desc.NodeID.String()},
+				Derivative: &deriv,
+			}
+			queries = append(queries, query)
+		}
+	}
+	now := time.Now().UnixNano()
+	tsReq := tspb.TimeSeriesQueryRequest{
+		EndNanos:    now,
+		StartNanos:  now - int64(40*time.Second),
+		SampleNanos: int64(10 * time.Second),
+		Queries:     queries,
+	}
+	tsResp, err := s.tsServer.Query(ctx, &tsReq)
+	if err != nil {
+		return nil, nil
+	}
+	for resIdx, tsResult := range tsResp.Results {
+		if len(tsResult.Datapoints) == 0 {
+			continue
+		}
+		nodeIdx := resIdx / len(IoMetrics)
+		metricIdx := resIdx % len(IoMetrics)
+		node := resp.Nodes[nodeIdx]
+		latestDataPoint := tsResult.Datapoints[len(tsResult.Datapoints)-1]
```

Comment:
While testing this, I inserted a println here: `latestDataPoint.TimestampNanos`. It seems that the value is always between 10 and 20 seconds in the past. Not quite sure why that is — seems like you could get a value <10 seconds in the past if the node summary has been written that recently.

Reply:
We don't allow queries for the currently in-progress time period because, when aggregating, it can cause visual issues with charts.  Even though you are not aggregating here, that behavior applies to all queries.